### ISSUE TITLE
Is there any imputation method in sktime?

### ISSUE THREAD

**Issue by zhenlingcn** [alan-turing-institute/sktime#603]
Currently, I'm encountering a prediction problem with some missing values. In the current implementation of sktime, when there is a missing value in the timing window, sktime will output nan. However, rather than tuning the size of the timing window, I prefer to fill these missing values. Thus, is there any method to do so?

**Comment by TonyBagnall** [comment]
hi, we do not currently have imputation methods in sktime that I am aware of, sorry. We would very much like to have some though!

**Comment by fkiraly** [comment]
You can get imputation through composition for any reduction method to a `sklearn` method, such as `DirectRegressionForecaster` - simply wrap a pipeline whose first element is a `sklearn` imputer, e.g., `SimpleImputer`.

Depending on your data format, you may also consider using `transformations.panel.interpolate` which can be used as imputer depending on the setting.

As @TonyBagnall, contributions are welcome - e.g., an imputation transformer based on the interpolator.

Let's maybe briefly also check with @mloning whether we didn't forget another potential solution to your problem.

**Comment by TonyBagnall** [comment]
thanks Franz, my knowledge of sktime is localised to the classification side, I am not really best place to comment .... so probably shouldnt :)

**Comment by mloning** [comment]
@zhenlingcn is this in the context of forecasting?

**Comment by zhenlingcn** [comment]
Yes, the context is forecasting. I can use pandas to simply fill in the missing values, such as filling in the missing values by mean value. However, the filling function provided by pandas is rather simple. In fact, I hope I can use more advanced filling methods, such as linear or non-linear interpolation.

**Comment by fkiraly** [comment]
Hm, I thought interpolators (padding, last value carry forward, linear/smoothing) were on the forecasting wishlist #220 or some roadmap, but I can't find them. Should definitely be there, since there are time series specific missing value handling strategies that you won't get from scikit-learn. @mloning, @TonyBagnall, do we have this recorded in some roadmap issue?

**Comment by mloning** [comment]
There is a related issue on handling missing values #69

**Comment by mloning** [comment]
Linear and non-linear interpolation are also provided by numpy, SciPy or pandas. See for example: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.interpolate.html

Since there isn't anything there, adding a simple `Interpolator` (series-to-series transformer) would be great! @zhenlingcn would you be interested in working on that? As a template take a look at the transformers here: https://github.com/alan-turing-institute/sktime/blob/master/sktime/transformations/series/acf.py

**Comment by aiwalter** [comment]
I am working on this!

**Comment by aiwalter** [comment]
@zhenlingcn solved by #637 . I think it can be closed then.
